File: test_ws.go
package main

import (
	"log"
	"net/url"
	"os"
	"os/signal"
	"time"

	"github.com/gorilla/websocket"
)

func main() {
	if len(os.Args) < 2 {
		log.Fatal("need benchmark ID")
	}
	u := url.URL{Scheme: "ws", Host: "localhost:8082", Path: "/ws/benchmarks/" + os.Args[1] + "/stream"}
	log.Printf("connecting to %s", u.String())

	c, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
	if err != nil {
		log.Fatal("dial:", err)
	}
	defer c.Close()

	done := make(chan struct{})

	go func() {
		defer close(done)
		for {
			_, message, err := c.ReadMessage()
			if err != nil {
				log.Println("read:", err)
				return
			}
			log.Printf("recv: %s", message)
		}
	}()

	interrupt := make(chan os.Signal, 1)
	signal.Notify(interrupt, os.Interrupt)

	select {
	case <-interrupt:
		log.Println("interrupt")
	case <-time.After(30 * time.Second):
		log.Println("timeout, exiting")
	}

	c.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
	<-done
}

File: test_ws.js
const WebSocket = require('ws');
const ws = new WebSocket('ws://localhost:8082/ws/benchmarks/91640f7d-cc4b-4238-a509-a4fc01110fe3/stream');

ws.on('open', function open() {
  console.log('connected');
});

ws.on('message', function incoming(data) {
  console.log('received: %s', data);
});

ws.on('error', function error(e) {
  console.log('error: ', e);
});
